pkg/capabilities/testing/mock_registry.go

```diff
@@ -0,0 +1,99 @@
+package testing
+
+import (
+	"context"
+	"errors"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+)
+
+type MockRegistry struct {
```

Comment:
Can we use mockery for this? Generate from CapabilitiesRegistry interface.

Reply:
Hmmm... I've gone ahead and used mockery for this but I'm actually not sure this is the right way to proceed. There are some tests which depend on a concrete registry implementation rather than on a mock, so that would leave us with some tests using the generated mock and others using this mockRegistry implementation.

I'm not wild on the fragmentation so was thinking of using the mock_registry everywhere. WDYT?

Reply:
How hard is it to convert all tests to use the generated mock? I'm fine with leaving it instead of generating one if it's actually useful, your call.

Reply:
Turns out not that hard 🤷 I've gone ahead and changed.